src/core/componentCreators.js

```diff
@@ -9,33 +9,55 @@ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTA
 OF ANY KIND, either express or implied. See the License for the specific language
 governing permissions and limitations under the License.
 */
+
 // This is the only place where core is allowed to import from components.
 // This makes sure that each component could be removed without breaking the library
+
 /* eslint-disable import/no-restricted-paths */
+
 import createDataCollector from "../components/DataCollector";
+/* @skipwhen ENV.alloy_activitycollector === false */
 import createActivityCollector from "../components/ActivityCollector";
 import createIdentity from "../components/Identity";
+/* @skipwhen ENV.alloy_audiences === false */
 import createAudiences from "../components/Audiences";
+/* @skipwhen ENV.alloy_personalization === false */
 import createPersonalization from "../components/Personalization";
 import createContext from "../components/Context";
 import createPrivacy from "../components/Privacy";
+/* @skipwhen ENV.alloy_eventmerge === false */
 import createEventMerge from "../components/EventMerge";
 import createLibraryInfo from "../components/LibraryInfo";
+/* @skipwhen ENV.alloy_decisioningengine === false */
 import createDecisioningEngine from "../components/DecisioningEngine";
+/* @skipwhen ENV.alloy_machinelearning === false */
 import createMachineLearning from "../components/MachineLearning";
 
-// TODO: Register the Components here statically for now. They might be registered differently.
-// TODO: Figure out how sub-components will be made available/registered
-export default [
-  createDataCollector,
-  createActivityCollector,
-  createIdentity,
-  createAudiences,
-  createPersonalization,
+const REQUIRED_COMPONENTS = [
```

Comment:
Can you add a comment here that the conditional build relies on the naming of this variable so don't change it.